asf_heat_pump_suitability/pipeline/flows/run_calculate_suitability_flow.py

```diff
@@ -0,0 +1,263 @@
+        )
+        from tqdm import tqdm
+        from asf_heat_pump_suitability.pipeline.suitability import calculate_suitability
+
+        print("Weighting scores and aggregating per LSOA")
+        self.weighted_scores = []
+
+        for lsoa_code in tqdm(self.input["lsoa"].drop_nulls().unique()):
+            self.weighted_scores.append(
+                calculate_suitability.aggregate_dict_lsoa_suitability_and_features(
+                    self.input, lsoa_code
+                )
+            )
+
+        self.next(self.concatenate_weighted_scores)
+
+    @step
+    def concatenate_weighted_scores(self, inputs):
+        """
+        Concatenate all weighted LSOA suitability scores together and filter to LSOAs with at least 15 records.
+        """
+        import polars as pl
+        import itertools
+
+        self.weighted_scores = list(
+            itertools.chain.from_iterable([input.weighted_scores for input in inputs])
+        )
+        logging.info(
+            "Filtering to LSOAs with data for at least 15 properties to be included in final dataset"
+        )
+        suitability_df = pl.DataFrame(self.weighted_scores).filter(
+            pl.col("n_properties") >= 15
+        )
+        self.suitability_df = suitability_df.with_columns(pl.col(pl.Float64).round(3))
+
+        self.next(self.join_proportion_of_flats_and_names)
+
+    @step
+    def join_proportion_of_flats_and_names(self):
+        """
+        Calculate the proportion of flats in each LSOA from the census data and join it to the suitability scores.
+        Join LSOA and DZ names to suitability scores.
+        """
+        import polars as pl
+        from asf_heat_pump_suitability.pipeline.prepare_features import (
+            property_type,
+            output_areas,
+        )
+
+        logging.info("Getting proportion of flats in each LSOA from the census data")
+        proportion_flats_df = (
+            property_type.transform_df_proportion_census_property_types()
+            .filter(pl.col("property_type") == "Flat, maisonette or apartment")
+            .select(["lsoa", "census_proportion"])
+            .rename({"census_proportion": "census_proportion_flats"})
+        )
+
+        logging.info("Getting LSOA & DZ names")
+        lsoa_names_df = output_areas.load_df_lsoa_dz_codes_names()
+
+        logging.info(
+            "Joining proportion of flats and LSOA & DZ names to suitability dataset"
+        )
+        self.suitability_df = self.suitability_df.join(
+            proportion_flats_df, how="left", on="lsoa"
+        ).join(lsoa_names_df, left_on="lsoa", right_on="lsoa_code", how="left")
+
+        self.next(self.save_outputs)
+
+    @step
+    def save_outputs(self):
+        """
+        Save outputs to S3.
+        """
+        from datetime import datetime
+        from asf_heat_pump_suitability.pipeline.utils import save_utils
```

Comment:
Think this is a bug, I get an error when I get to this step and is inconsistent with the rest of our steps


```suggestion
        from asf_heat_pump_suitability.utils import save_utils
```

Reply:
hmmm.. this should not happen. It might be an artefact of the `sym-linking` i did. I will look into it.

Reply:
I have looked into this. I cannot see that git was tracking the symlink so I'm not sure why this happened for you. Here's the commands I ran in terminal:

1. `find . -type l` - this shows me there is a symlink in my local copy of the repo in `asf_heat_pump_suitability/pipeline/utils`. 
2. `find . -type l -exec ls -ld {} +` - this shows me my local symlink is linking the above folder to `asf_heat_pump_suitability/utils`
3. `git rm asf_heat_pump_suitability/pipeline/utils` - this should remove the file from git's tracking. It outputted an error: `fatal: pathspec 'asf_heat_pump_suitability/pipeline/utils' did not match any files`. This means it's not being tracked by git.
4. `git ls-files -s | grep "utils"` - this lists the paths that git is tracking with the word 'utils' in it. I can see that `'asf_heat_pump_suitability/pipeline/utils` is not present in this list, confirming git is not tracking this directory.
5. `rm asf_heat_pump_suitability/pipeline/utils` - this removes the symlink directory from my local repo
6. `git status` - i can see there are no changes/deletions to commit.

From this, it seems like the symlinking was local only but your fix implies that your local repo also has the symlink. I would suggest pulling the branch again and then running the command in step 1 above to see if you have a local symlink and let me know.

Reply:
🤦‍♀️ ok now i see that I read your change the wrong way round. Ignore all my comments. What you've done is correct and has fixed an actual bug!!! bro whyyyy am i so incapable of reading